cookbooks/aws-parallelcluster-slurm/recipes/update_head_node.rb

```diff
@@ -44,12 +43,39 @@ def get_slurm_nodelist(queue)
     node_list
   end
 
-  def update_nodes(power_saving, nodelist)
-    command = "sudo -i scontrol update state=#{power_saving} nodename=#{nodelist}"
+  def update_slurm_nodes(state, nodelist)
+    command = "sudo -i scontrol update state=#{state} nodename=#{nodelist} reason='updating node state during cluster update'"
     Chef::Log.info("Executing node state update with command (#{command})")
     execute_command(command)
   end
 
+  def split_static_and_dynamic_nodes(nodelist)
+    static_node = []
+    dynamic_node = []
+    nodes = nodelist.split(',')
+    nodes.each do |node|
+      if node.include?('-st-')
```

Comment:
I'd use the `hit_is_static_node?` in the `helpers.rb`, it will automatically raise an exception if the nodename format is wrong.

```ruby
hit_is_static_node ? static_nodes.push(node) : dynamic_nodes.push(node)
```

Reply:
changed